awsbestpractices/mutually-exclusive-method-calls/MutuallyExclusiveMethodCalls.java

```diff
@@ -0,0 +1,71 @@
+package example;
+
+import com.amazonaws.services.kms.model.GenerateDataKeyRequest;
+
+public class Example {
+    // Based on
+    // https://code.amazon.com/packages/PinguServiceCrypto/blobs/9eb45868edb542dc8d042a4da18ad91f137593c2/--/src/com/amazon/kms/pingu/crypto/PinguKeystoreCrypto.java#L709-L710
+    // Must flag
+    private GenerateDataKeyRequest setKeySpecAndNumberOfBytes(String keySpec, Integer numberOfBytes) {
+        GenerateDataKeyRequest generateRequest = new GenerateDataKeyRequest();
+        generateRequest.setKeySpec(keySpec);
+        generateRequest.setNumberOfBytes(numberOfBytes);
+        return generateRequest;
+    }
+
+    // Must flag
+    private GenerateDataKeyRequest withKeySpecAndNumberOfBytes(String keySpec, Integer numberOfBytes) {
+        GenerateDataKeyRequest generateRequest = new GenerateDataKeyRequest()
+                .withNumberOfBytes(numberOfBytes)
+                .withFoo("bar")
+                .withKeySpec(keySpec);
+        return generateRequest;
+    }
+
+    // Must flag
+    private void multipleDetections(String keySpec, Integer numberOfBytes) {
+        GenerateDataKeyRequest generateRequest1 = new GenerateDataKeyRequest();
+        generateRequest1.setKeySpec(keySpec);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code sets two parameters that cannot be used together. Consider eliminating one of the parameters. Parameter names: setKeySpec, setNumberOfBytes. Refer to [the documentation](https://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateDataKey.html) for details.